numbat/src/datetime.rs

```diff
@@ -1,82 +1,65 @@
-use chrono::{DateTime, Datelike, FixedOffset, LocalResult};
-use chrono_tz::Tz;
+use jiff::{civil::DateTime, fmt::rfc2822, tz::TimeZone, Zoned};
 
-#[cfg(feature = "local-timezone")]
-pub fn get_local_timezone() -> Option<Tz> {
-    let tz_str = iana_time_zone::get_timezone().ok()?;
-    tz_str.parse().ok()
+pub fn get_local_timezone_or_utc() -> TimeZone {
+    TimeZone::system()
 }
 
-#[cfg(feature = "local-timezone")]
-pub fn get_local_timezone_or_utc() -> Tz {
-    get_local_timezone().unwrap_or(chrono_tz::UTC)
-}
-
-#[cfg(not(feature = "local-timezone"))]
-pub fn get_local_timezone_or_utc() -> Tz {
-    chrono_tz::UTC
-}
-
-pub fn parse_datetime(input: &str) -> Option<DateTime<FixedOffset>> {
-    if let Ok(dt) = chrono::DateTime::parse_from_rfc3339(input) {
-        Some(dt)
-    } else if let Ok(dt) = chrono::DateTime::parse_from_rfc2822(input) {
+pub fn parse_datetime(input: &str) -> Option<Zoned> {
+    if let Ok(dt) = DateTime::strptime("%Y-%m-%dT%H:%M:%SZ", input) {
+        // RFC 3339
+        Some(dt.to_zoned(TimeZone::UTC).unwrap()) // TODO
+    } else if let Ok(dt) = rfc2822::parse(input) {
         Some(dt)
     } else {
         const FORMATS: [&str; 8] = [
             // 24 hour formats:
-            "%Y-%m-%d %H:%M:%S%.f",
-            "%Y/%m/%d %H:%M:%S%.f",
+            "%Y-%m-%d %H:%M:%S", // TODO: add support for fractional seconds
+            "%Y/%m/%d %H:%M:%S", // TODO: add support for fractional seconds
             "%Y-%m-%d %H:%M",
             "%Y/%m/%d %H:%M",
             // 12 hour formats:
-            "%Y-%m-%d %I:%M:%S%.f %p",
+            "%Y-%m-%d %I:%M:%S %p", // TODO: add support for fractional seconds
             "%Y-%m-%d %I:%M %p",
-            "%Y/%m/%d %I:%M:%S%.f %p",
+            "%Y/%m/%d %I:%M:%S %p", // TODO: add support for fractional seconds
             "%Y/%m/%d %I:%M %p",
         ];
 
         for format in FORMATS {
             // Try to match the given format plus an additional UTC offset (%z)
-            if let Ok(dt) = chrono::DateTime::parse_from_str(input, &format!("{format} %z")) {
+            if let Ok(dt) = Zoned::strptime(&format!("{format} %z"), input) {
                 return Some(dt);
             }
 
             // Try to match the given format plus an additional timezone name (%Z).
-            // chrono does not support %Z, so we implement this ourselves. We were
+            // jiff does not support %Z, so we implement this ourselves. We were
```

Comment:
Is is hard to support `%Z` when parsing datetimes? I know that chrono doesn't support this either, but it seems like `jiff` might be able to, since there is a tighter coupling to timezones and the timezone database?

Reply:
It can't really be done, because the time zone abbreviations are ambiguous. `CST` for example is an abbreviation for `Central Standard Time`, `China Standard Time` _and_ `Cuba Standard Time`. So there's just no way to parse it in a way that doesn't lead to a footgun.

The better way is to parse IANA time zone identifiers. The only way Jiff supports that currently is via the RFC3339/RFC9557/ISO-8601 format that looks like this: `2024-07-28T09:31-04[America/New_York]`. The `strtime` module doesn't have a directive for parsing/formatting an IANA time zone identifier, _yet_. I don't think there is any specific roadblock to adding that support, but AIUI, there isn't a precedent for it in strtime APIs in other ecosystems. So I was hesitant to add it.

Reply:
> It can't really be done, because the time zone abbreviations are ambiguous. `CST` for example is an abbreviation for `Central Standard Time`, `China Standard Time` _and_ `Cuba Standard Time`.

Right, that is unfortunate. As a user, I would *still* like to have that functionality for cases where it's not ambiguous. I would be okay with not being able to parse `2023-07-28 09:00:00 CST`, but I would like to be able to parse [`2023-07-28 09:00:00 US/Central`](https://numbat.dev/?q=datetime%28%222023-07-28+09%3A00%3A00+US%2FCentral%22%29%E2%8F%8E).

> So there's just no way to parse it in a way that doesn't lead to a footgun.

I'm definitely not saying this is important. But for our use case, it would certainly be great if I could just use `%Z` in a datetime format and get a result *if it is unambiguous*, and an error otherwise.

> The better way is to parse IANA time zone identifiers. The only way Jiff supports that currently is via the RFC3339/RFC9557/ISO-8601 format that looks like this: `2024-07-28T09:31-04[America/New_York]`. The `strtime` module doesn't have a directive for parsing/formatting an IANA time zone identifier, _yet_. I don't think there is any specific roadblock to adding that support, but AIUI, there isn't a precedent for it in strtime APIs in other ecosystems. So I was hesitant to add it.

Having a new/special `%`-identifier for that would be great, but again — we already have something that works ok for us, so please don't add this unless others find this useful as well.

Reply:
All righty. When I get a chance, I'll make an issue for this on Jiff and work through some of the choices here.

And yeah, `US/Central` is an IANA time zone identifier. So if `strptime` had a switch for those, then `US/Central` would work. As would `America/New_York` among many others.